eox_core/middleware.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/python
+"""
+This file implements the Middleware support for the Open edX platform.
+A microsite enables the following features:
+1) Mapping of sub-domain name to a 'brand', e.g. foo-university.edx.org
+2) Present a landing page with a listing of courses that are specific to the 'brand'
+3) Ability to swap out some branding elements in the website
+"""
+import logging
+import re
+import six
+
+from django.conf import settings
+from django.contrib.auth import REDIRECT_FIELD_NAME
+from django.contrib.auth.views import redirect_to_login
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from django.http import Http404, HttpResponseRedirect
+
+from eox_core.edxapp_wrapper.configuration_helpers import get_configuration_helper
+from eox_core.models import Redirection
+
+from eox_core.utils import cache, fasthash
+
+
+configuration_helper = get_configuration_helper()  # pylint: disable=invalid-name
+
+HOST_VALIDATION_RE = re.compile(r"^(?:[0-9]{1,3}\.){3}[0-9]{1,3}(:[0-9]{2,5})?$")
```

Comment:
I think this var `HOST_VALIDATION_RE` is not used across this code (used in eox-tenant)